### PR TITLE
Use strategic patch to replace changeCause in patch command

### LINKED ISSUE #24858: kubectl patch --record doesn't show up at rollout history
Hi,

Deployment patching with `--record` doesn't show up in the deployment rollout history:

first:
```
kubectl patch deployment mydep -p '{"spec":{"template":{"spec":{"containers":[{"name":"mycontainer","image":"gcr.io/project/my-container:TAG_NAME"}]}}}}' --record
```

after:
```
kubectl rollout history deployment/mydep
```
```
REVISION	CHANGE-CAUSE
1		<none>
10		<none>
2		<none>
3		<none>
4		<none>
```

Is it a bug? Should it show up?

Thanks,
Peter

### PR DESCRIPTION
**Pull request by dshulyak**
This is partial rework of 11da9a7638e393f84eb34fcf54c1962b3cdc115c
StrategicPatch will be used to update changeCause but failure wont affect command result

fixes: #24858

### CODE REVIEW

**Comment by dshulyak** [comment]
@deads2k May I ask for review?

File: pkg/kubectl/cmd/patch.go [hunk 1]
```diff
@@ -150,17 +150,14 @@ func RunPatch(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []stri
...
 		if cmdutil.ShouldRecord(cmd, info) {
-			if err := cmdutil.RecordChangeCause(patchedObject, f.Command()); err == nil {
-				// don't return an error on failure.  The patch itself succeeded, its only the hint for that change that failed
-				// don't bother checking for failures of this replace, because a failure to indicate the hint doesn't fail the command
-				// also, don't force the replacement.  If the replacement fails on a resourceVersion conflict, then it means this
-				// record hint is likely to be invalid anyway, so avoid the bad hint
-				resource.NewHelper(client, mapping).Replace(namespace, name, false, patchedObject)
+			patch, err := cmdutil.ChangeResourcePatch(info, f.Command())
+			if err == nil {
+				helper.Patch(info.Namespace, info.Name, api.StrategicMergePatchType, patch)
```
Comment (deads2k) [inline pkg/kubectl/cmd/patch.go]: Add a comment about suppressing the failure and why its good to do that.

**Comment by janetkuo** [comment]
ok to test

**Comment by deads2k** [comment]
@dshulyak  probably a flake, rebase on master and re-push to kick it.  I didn't see anything too obvious in the issue list, so it may already be resolved.

**Pull request merged**